Utils/src/PDFWeightProducer.cc

```diff
@@ -201,6 +213,70 @@ void PDFWeightProducer::produce(edm::StreamID, edm::Event& iEvent, const edm::Ev
       }
       found_pdfs = true;
     }
+
+    //if rf scale weights still not found, recalculate them
+    //based on https://gitlab.cern.ch/cms-desy-top/TopAnalysis/-/blob/Htottbar_2016/ZTopUtils/plugins/EventWeightMCSystematicRecalc.cc
+    if(!found_scales and recalculateScales_){
+      double kUp = 2;
+      double kDn = 0.5;
+      float Q = genHandle->pdf()->scalePDF;
+
+      //factorization scale
+      int id1 = genHandle->pdf()->id.first;
+      double x1 = genHandle->pdf()->x.first;
+
+      int id2 = genHandle->pdf()->id.second;
+      double x2 = genHandle->pdf()->x.second;
+
+      double pdf1, pdf1up, pdf1dn;
+      double pdf2, pdf2up, pdf2dn;
+      {
+         std::lock_guard<std::mutex> lock(mutex_);
+         LHAPDF::usePDFMember(1,0);
+         pdf1 = LHAPDF::xfx(1, x1, Q, id1)/x1;
+         pdf2 = LHAPDF::xfx(1, x2, Q, id2)/x2;
+         pdf1up = LHAPDF::xfx(1, x1, kUp*Q, id1)/x1;
+         pdf2up = LHAPDF::xfx(1, x2, kUp*Q, id2)/x2;
+         pdf1dn = LHAPDF::xfx(1, x1, kDn*Q, id1)/x1;
+         pdf2dn = LHAPDF::xfx(1, x2, kDn*Q, id2)/x2;
+      }
+      //for debugging
+      //std::cout << "pdf1 = " << pdf1 << ", pdf1up = " << pdf1up << ", pdf1dn = " << pdf1dn << ", pdf2 = " << pdf2 << ", pdf2up = " << pdf2up << ", pdf2dn = " << pdf2dn << std::endl;
```

Comment:
It would be better to add a `debug` flag and then use an `edm::MessageLogger` instance. That way you don't need to keep this comment around.